### PR TITLE
Error reporting [Example] [Do not merge]

### PR DESCRIPTION
**Pull request by konstantinzolotarev**

### CODE REVIEW

**Comment by konstantinzolotarev** [comment]
Related to #488

File: lib/server.js [hunk 1]
```diff
@@ -391,28 +397,34 @@ Server.prototype.onWebSocket = function (req, socket) {
...
 
       // transport error handling takes over
       socket.removeListener('error', onUpgradeError);
 
-      var transport = new transports[req._query.transport](req);
-      if (req._query && req._query.b64) {
-        transport.supportsBinary = false;
-      } else {
-        transport.supportsBinary = true;
+      try {
```
Comment (darrachequesne) [inline lib/server.js]: Is this `try..catch` needed here? Is there a particular issue?

**Comment by darrachequesne** [comment]
Is the `debugError` actually needed in `lib\server.js`? I think it might make sense in `lib\socket.js`, in order to split received/sent messages (how about naming it `trace()`?) from transport details. What do you think?

File: lib/server.js (hunk 1 above)
Comment (konstantinzolotarev) [inline lib/server.js]: I was not sure but in https://github.com/socketio/engine.io/blob/master/lib/server.js#L299 
Same statement wrapped in `try..catch` so I did same.

**Comment by konstantinzolotarev** [comment]
@darrachequesne Yep make sense. Will continue then. This PR Could be closed.

**Pull request closed without merging**